vignettes/test-children.Rmd

```diff
@@ -0,0 +1,181 @@
+  but rcrunch has tools that try to make this as easy as possible.
+
+
+## Integration testing strategy
+We do not advise that child packages run integration tests. If you feel you need them, 
+please contact the Crunch team and we may be able to grant you access to the 
+Crunch.io test server and help you write tests that use that.
+
+
+## Mock testing strategy
+Mock tests for the crunch package are tested using the `httptest` package, which was 
+written while the author worked at Crunch.io and so is tailored to our situation quite
+well. Before continuing, I highly recommend reading the vignettes for that package:
+
+- [`httptest`: A Test Environment for HTTP Requests in R](https://enpiar.com/r/httptest/articles/httptest.html)
+- [Redacting and Modifying Recorded Requests](https://enpiar.com/r/httptest/articles/redacting.html)
+
+Another strategy for generating mock tests is to use the 
+[`mockr` packages](https://krlmlr.github.io/mockr/) (this function used to live 
+in `testthat` itself, but will be deprecated in the next version of `testthat` at 
+the time of writing this). 
+
+The `with_mock()` approach has the advantage that there are fewer moving parts to go
+wrong. However, this approach has the disadvantage that it allows your mocks to be 
+completely divorced from the crunch API and rcrunch package. Therefore it will be harder
+for your tests to get updated in the event that the behavior of rcrunch changes.
+
+
+### Step 1: Capture requests (ideally, reproducibly)
+The `httptest` package has functions `start_capturing()` and `stop_capturing()`
+that allow it to save the results of API calls that you make in 
+a way that it will be able to replay later, while you're testing. You'll
+want to capture all the requests needed to run the test and save them
+in the `testthat` folder of your package so that your tests have access to
+them. If you've already written the test & function, you can run that same
+code inside of `httptest`'s capturing mechanism. However, if you're trying
+to write your tests before you've written the function, you'll need to 
+plan out what `crunch` functions you will need to run so that you can
+capture them.
+
+It is a really good idea to save this code in a reproducible state
+so that you can rerun it if your code changes, or if the crunch API 
+changes. To make it reproducible, I recommend:
+
+1) Saving the script in your package, a good place is in the `testthat`
+   folder, at the level above where the fixtures are saved.
+2) Deleting existing mock files if they already exist.
+3) Either creating the dataset in the script, or at least starting from 
+   a dataset in a project stored in a place that everyone who works on
+   the package have access to.
+
+Putting that altogether, here's an example of a script. It could be saved
+for example in a file `"tests/testthat/captured/generate_1.R"`.
+
+Before running, it's generally best to start from a clean R session
```

Comment:
@1beb - couldn't bring myself to directly recommend `rm(list = ls())`, but mention it in text here

Reply:
Mulligan!